airflow/providers/openlineage/utils/utils.py

```diff
@@ -155,6 +155,14 @@ def is_selective_lineage_enabled(obj: DAG | BaseOperator | MappedOperator) -> bo
         raise TypeError("is_selective_lineage_enabled can only be used on DAG or Operator objects")
 
 
+def extract_nested_task_group_id(task):
```

Comment:
There can be cases where task_id in the TaskGroup is just the task_id (not task group prefix). So how does this work with `prefix_group_id=False`?
https://github.com/apache/airflow/blob/cd69f8efc0e26507518757ed4a357cba74cc8a13/airflow/utils/task_group.py#L103


Please also add unit test to cover this change